test/sharness/lib/iptb-lib.sh

```diff
@@ -26,6 +26,17 @@ iptb() {
         esac
         return 1
     fi
+
+    # If we are looking for clean-start, issue a 'stop' twice for good measure
+    # It sadly doesn't always work :/
+    if test "$GOIPFSTEST_ENSURE_CLEAN_START" = 1 \
+    && test "$#" = 1 \
+    && test "$1" = stop \
+    ; then
+      sleep 1
+      command iptb stop > /dev/null 2>&1 &
+      sleep 1
+    fi
```

Comment:
`iptb stop` _should_ wait, and calling stop a second time shouldn't help. Are you sure we don't just have leftovers from a previous failed run?

IPTB currently:

1. Sends sigterm and waits a second.
2. Sends sigterm again and waits 2 seconds.
3. Sends sigquit and waits 5 seconds.
4. Sends sigkill and waits 5 seconds.

Reply:
99% this is not the case, but I will try to instrument this better and get back to this.

Let's hold off merging this until I had a moment to investigate the iptb-specific stuff more in depth.

Reply:
The code is https://github.com/ipfs/iptb-plugins/blob/d63cc1c88acb3195428f633bba332a5f5f53a8ea/local/localipfs.go#L203 (might be buggy).

Reply:
@ribasushi could we pull out everything unrelated to the clean start logic so we can merge that separately?

(stuff sitting in PRs tends to die)

Reply:
> stuff sitting in PRs tends to die

@Stebalien not my stuff as a general rule, but yes, will separate into "less contentious" piece today-ish ( I am traveling to congress, will be in-and-out )

Reply:
@Stebalien https://github.com/ipfs/go-ipfs/pull/6805 is now smoking, I am still keeping this PR open to fix the remaining parts, ETA some time this year